part-2.py

```diff
@@ -4,13 +4,34 @@
 # the appropriate comment.
 
 # search
-
+def search(array, query):
+    if not len(array):
+        return False
+    elif array[0] == query:
+        return True
+    return search(array[1:], query)
 
 
 # is_palindrome
-
+def is_palindrome(text):
+    return text == text[::-1]
 
 
 # digit_match
-
+def digit_match(apples, oranges):
+    # Base cases:
+    # If both apples and oranges are 0 return 1
+    if apples == 0 and oranges == 0:
```

Comment:
I had to do some thinking and debugging to understand why it works to always return 1 for this case but it does! Great work!